modules/apps/item-selector/item-selector-web/src/main/java/com/liferay/item/selector/web/internal/portlet/LocalizedItemSelectorRendering.java

```diff
@@ -59,7 +59,15 @@ public void add(ItemSelectorViewRenderer itemSelectorViewRenderer) {
 
 		String title = itemSelectorView.getTitle(_locale);
 
-		_itemSelectorViewRenderers.put(title, itemSelectorViewRenderer);
+		if (_itemSelectorViewRenderers.put(title, itemSelectorViewRenderer) !=
+				null) {
+
+			_navigationItems.removeIf(
+				navigationItem -> title.equals(
+					navigationItem.get(
+						"label"
+					).toString()));
+		}
```

Comment:
Can you split the update of the data structure from the check?
```
ItemSelectorViewRenderer previousItemSelectorViewRenderer = _itemSelectorViewRenderers.put(...);

if (previousItemSelectorViewRenderer != null) {
    ...
}

Reply:
Also, avoid chaining method calls. In
```
					navigationItem.get(
						"label"
					).toString()
```
replace the toString() call with `String.valueOf(navigationItem.get("label"))`.